### PR TITLE
[WFLY-15257] Update the descriptions in LocalDescriptions.properties for the Elytron OIDC client subsystem

### PR DESCRIPTION
**Pull request by petrberan**
https://issues.redhat.com/browse/WFLY-15257

### CODE REVIEW

**Comment by petrberan** [comment]
Tagging @fjuma and @theashiot for proofreading. Please don't merge until @fjuma approves the changes

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (fjuma) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:4]: I don't think the `realm` element config can be copied from the Keycloak Admin Console (only the configuration for a specific client could be copied).

Here, maybe we can update the description to something like:
Configuration for a Keycloak realm. If you have multiple deployments secured by the same realm you can share the realm configuration here.

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (fjuma) [reply elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:10]: s/when Keycloak rotate/when Keycloak rotates

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (fjuma) [reply elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:46]: s/in the PEM/in PEM

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (fjuma) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:47]: Can also add:

It is recommended to use the `provider-url` instead.

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (fjuma) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:81]: s/deployment to be secured/deployment secured

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (fjuma) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:82]: s/realm used/realm to be used

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (fjuma) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:84]: s/provider realm/provider

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (fjuma) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:85]: Could add:

It is recommended to use `provider-url` instead.

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (fjuma) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:95]: This is the client identifier for the application. This is the identifier that has been registered with the authorization server.

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (fjuma) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:94]: This is the client identifier for the application. This is the identifier that has been registered with the authorization server.

**Review by fjuma — CHANGES_REQUESTED** [review]
Thanks very much for working on this @petrberan! I've added some comments.

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:104]: s / keystore / key store
s / client-keystore / 'client-keystore'

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:105]: s / client-keystore / 'client-keystore'

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:106]: s / the Cross-Origin / Cross-Origin
s / the value for / the value of

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:107]: s / the Cross-Origin / Cross-Origin

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:108]: s / the Cross-Origin / Cross-Origin

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:109]: s / the Cross-Origin / Cross-Origin

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:119]: How about:

"If the subsystem recognizes a token signed by an unknown public key, it will try to download a new public key from the server. The subsystem won't try to download a public key if it already tried last in less than 'min-time-between-jwks-requests' seconds."

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:120]: How about:

"The maximum interval between two requests to retrieve new public keys. New public keys are downloaded when the subsystem recognizes a token signed by an unknown public key. Even if the token's key is already known, new public keys are downloaded periodically as per the interval set here."

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:123]: s / , verifies / , the subsystem verifies
s /  set to 'true' / set the value to 'true'

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:127]: s / credentials / the credentials

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:128]: s / client-keystore-file / 'client-keystore-file'

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:129]: s / keystore / key store
s / client-keystore-file / 'client-keystore-file'

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:130]: s / keystore / key store

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:131]: s / keystore / key store

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:135]: s / Specify the credential / The credential

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:139]: s / , in milliseconds / in milliseconds
s / , negative value / , and a negative

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [inline elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:141]: s / , negative value / , and a negative value

**Review by theashiot — COMMENTED** [review]
Hi @petrberan, thanks for the updates, I've added some comments.

cheers,
ashwin

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (petrberan) [reply elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:54]: I believe the truststore here refers to the Java truststore (or properly even Java TrustStore), which is written without a space. Same for the keystore (Java KeyStore), see https://docs.oracle.com/cd/E19509-01/820-3503/6nf1il6er/index.html

I went with lowercase names for the consistency with the rest of the codebase, where the lowercase truststore and keystore are more common than KeyStore and TrustStore

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (petrberan) [reply elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:4]: I believe this part was copied from here: https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/8-beta/html/using_single_sign-on_with_jboss_eap/ref_elytron-oidc-client-subsystem-attributes_default 

I'm not sure how accurate the description in the link is, but I've changed this one as advised.

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (theashiot) [reply elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:54]: thanks, @petrberan in that case please ignore my comments about keystore -> key store and truststore -> trust store globally.

File: elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties
Comment (petrberan) [reply elytron-oidc-client/subsystem/src/main/resources/org/wildfly/extension/elytron/oidc/LocalDescriptions.properties:30]: Those words in <> are suitable just right. I added a warning to better clarify what the parameter does. I'll resolve this and send the changed version in new commit

**Comment by petrberan** [comment]
Thanks for your help @fjuma @theashiot . I've tried to process all your comments and change the lines accordingly. 

Can you have a look again @fjuma and approve this PR if the changes are alright?

**Review by fjuma — APPROVED** [review]
Thanks @petrberan!

**Comment by bstansberry** [comment]
Thanks @petrberan, @theashiot and @fjuma!

**Pull request merged**
